src/styles/main.scss

```diff
@@ -1,3 +1,128 @@
+@import "var.scss";
+
+* {
+  box-sizing: border-box;
+}
+
 body {
   margin: 0;
 }
+
+.calendar {
+  margin-top: 150px;
+  position: relative;
+  left: 50%;
+  transform: translate(-50%);
+  max-width: 706px;
+  display: flex;
+  gap: 1px;
+  flex-wrap: wrap;
+
+  &--start-day-sun .calendar__item:first-child {
+    margin-left: $sun;
+  }
+
+  &--start-day-tue .calendar__item:first-child {
+    margin-left: $tue;
+  }
+
+  &--start-day-wed .calendar__item:first-child {
+    margin-left: $wed;
+  }
+
+  &--start-day-thu .calendar__item:first-child {
+    margin-left: $thu;
+  }
+
+  &--start-day-fri .calendar__item:first-child {
+    margin-left: $fri;
+  }
+
+  &--start-day-sat .calendar__item:first-child {
+    margin-left: $sat;
+  }
+
+  &--start-day-mon .calendar__item:first-child {
+    margin-left: $mon;
+  }
+
+  &--month-lenght--29 {
```

Comment:
return your old loop from `1` to `31` where you add `content` by `index`. create another loop from `28` to `31` where you select `.calendar--month-length--#{$index}` and hide blocks starting from `index`